### ISSUE TITLE
kubectl describe job should print key fields for scheduling (suspend, tolerations, nodeSelector)

### ISSUE THREAD

**Issue by mimowo** [kubernetes/kubernetes#122613]
### What would you like to be added?

kubectl describe job is missing some key fields. For example, for yaml:
```
apiVersion: batch/v1
kind: Job
metadata:
  name: sample-job
  namespace: default
spec:
  parallelism: 1
  completions: 1
  suspend: true
  template:
    spec:
      tolerations:
      - key: "nvidia.com/gpu"
        operator: "Exists"
        effect: "NoSchedule"
      nodeSelector:
        cloud.google.com/gke-accelerator: nvidia-tesla-t4
      containers:
      - name: dummy-job
        image: gcr.io/k8s-staging-perf-tests/sleep:v0.0.3
        args: ["1800s"]
        resources:
          requests:
            cpu: "100m"
            memory: "100Mi"
            nvidia.com/gpu: 1
          limits:
            nvidia.com/gpu: 1
      restartPolicy: Never
```
`kubectl describe job` returns:

```
Name:             sample-job
Namespace:        default
Selector:         batch.kubernetes.io/controller-uid=6fb9fe7f-bb73-4e2c-a361-71210724430f
Labels:           batch.kubernetes.io/controller-uid=6fb9fe7f-bb73-4e2c-a361-71210724430f
                  batch.kubernetes.io/job-name=sample-job
                  controller-uid=6fb9fe7f-bb73-4e2c-a361-71210724430f
                  job-name=sample-job
Annotations:      <none>
Parallelism:      1
Completions:      1
Completion Mode:  NonIndexed
Pods Statuses:    0 Active (0 Ready) / 0 Succeeded / 0 Failed
Pod Template:
  Labels:  batch.kubernetes.io/controller-uid=6fb9fe7f-bb73-4e2c-a361-71210724430f
           batch.kubernetes.io/job-name=sample-job
           controller-uid=6fb9fe7f-bb73-4e2c-a361-71210724430f
           job-name=sample-job
  Containers:
   dummy-job:
    Image:      gcr.io/k8s-staging-perf-tests/sleep:v0.0.3
    Port:       <none>
    Host Port:  <none>
    Args:
      1800s
    Limits:
      nvidia.com/gpu:  1
    Requests:
      cpu:             100m
      memory:          100Mi
      nvidia.com/gpu:  1
    Environment:       <none>
    Mounts:            <none>
  Volumes:             <none>
Events:
  Type    Reason     Age   From            Message
  ----    ------     ----  ----            -------
  Normal  Suspended  4s    job-controller  Job suspended
```
**Issue**: some important fields are missing such as `suspend`, `nodeSelector` and `tolerations`.

For comparison, the `kubectl describe pod` contains the relevant information:
```
Name:             sample-job-zkz47
Namespace:        default
Priority:         0
Service Account:  default
Node:             <none>
Labels:           batch.kubernetes.io/controller-uid=6fb9fe7f-bb73-4e2c-a361-71210724430f
                  batch.kubernetes.io/job-name=sample-job
                  controller-uid=6fb9fe7f-bb73-4e2c-a361-71210724430f
                  job-name=sample-job
Annotations:      <none>
Status:           Pending
IP:               
IPs:              <none>
Controlled By:    Job/sample-job
Containers:
  dummy-job:
    Image:      gcr.io/k8s-staging-perf-tests/sleep:v0.0.3
    Port:       <none>
    Host Port:  <none>
    Args:
      1800s
    Limits:
      nvidia.com/gpu:  1
    Requests:
      cpu:             100m
      memory:          100Mi
      nvidia.com/gpu:  1
    Environment:       <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-44x7m (ro)
Conditions:
  Type           Status
  PodScheduled   False 
Volumes:
  kube-api-access-44x7m:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              cloud.google.com/gke-accelerator=nvidia-tesla-t4
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
                             nvidia.com/gpu:NoSchedule op=Exists
Events:
  Type     Reason            Age   From                                   Message
  ...
```

### Why is this needed?

The mentioned fields are important for job scheduling for projects like [Kueue](https://github.com/kubernetes-sigs/kueue).

The `kubectl describe` command is often used for debugging purposes, but its output might be misleading. For example, seeing the output one may assume the job does not have tolerations or nodeSelector, but it is not possible to tell based on the output.

**Comment by mimowo** [comment]
cc @alculquicondor 
/sig cli
/sig apps

**Comment by mimowo** [comment]
cc @kaisoz

**Comment by kumarankit999** [comment]
Thanks @mimowo for raising this issue, it would be great if you can refer any link of that page by which you have raised this Issue!! It will much help to new contributors!!

**Comment by alculquicondor** [comment]
Here is where the code lives https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/kubectl/pkg/describe

**Comment by raminderis** [comment]
Why is the fix only appearing in 1.26.15.

I always thought fix will appear in the most recent i.e. 1.29 as a patch version first, and then backward ported to other supported releases. isnt that true?

**Comment by alculquicondor** [comment]
Only critical bugfixes should be backported. The fix will actually be available in kubectl v1.30 and newer.
